File: Lab_2/input/input2.ts
interface FilmRecord {
  filmId: number;
  filmTitle: string;
  filmMaker: string;
  productionYear: number;
  scoreValue: number;
}

interface PerformerData {
  performerId: number;
  performerName: string;
  performerAge: number;
  filmographyCount: number;
}

function generateFilm(
  filmIdentifier: number,
  motionPictureTitle: string,
  directorName: string,
  yearOfRelease: number
): FilmRecord {
  return {
    filmId: filmIdentifier,
    filmTitle: motionPictureTitle,
    filmMaker: directorName,
    productionYear: yearOfRelease,
    scoreValue: 0
  };
}

function generatePerformer(
  performerIdentifier: number,
  actorName: string,
  actorAge: number
): PerformerData {
  return {
    performerId: performerIdentifier,
    performerName: actorName,
    performerAge: actorAge,
    filmographyCount: 0
  };
}

function modifyScore(
  filmData: FilmRecord,
  updatedScore: number
): FilmRecord {
  return {
    ...filmData,
    scoreValue: updatedScore
  };
}

const checkIfOldFilm = (filmItem: FilmRecord): boolean => filmItem.productionYear < 1980;
const verifyAdultPerformer = (performerItem: PerformerData): boolean => performerItem.performerAge >= 18;

const outputFilmTitle = (filmRecord: FilmRecord): void => {
  console.log(filmRecord.filmTitle);
};

const outputPerformerDetails = (actorRecord: PerformerData): void => {
  console.log(`Performer: ${actorRecord.performerName}, Age: ${actorRecord.performerAge}`);
};

function incrementFilmCount(actorInfo: PerformerData): PerformerData {
  return {
    ...actorInfo,
    filmographyCount: actorInfo.filmographyCount + 1
  };
}

File: Lab_2/output/input2.py
@dataclass
class FilmRecord:
    filmId: float
    filmTitle: str
    filmMaker: str
    productionYear: float
    scoreValue: float

@dataclass
class PerformerData:
    performerId: float
    performerName: str
    performerAge: float
    filmographyCount: float

def generateFilm(filmIdentifier: float, motionPictureTitle: str, directorName: str, yearOfRelease: float) -> FilmRecord:
    return {
    filmId: filmIdentifier,
    filmTitle: motionPictureTitle,
    filmMaker: directorName,
    productionYear: yearOfRelease,
    scoreValue: 0
    }

def generatePerformer(performerIdentifier: float, actorName: str, actorAge: float) -> PerformerData:
    return {
    performerId: performerIdentifier,
    performerName: actorName,
    performerAge: actorAge,
    filmographyCount: 0
    }

def modifyScore(filmData: FilmRecord, updatedScore: float) -> FilmRecord:
    filmData.scoreValue = updatedScore
    return filmData

checkIfOldFilm = lambda filmItem: filmItem.productionYear < 1980  # type: (FilmRecord) -> bool

def outputFilmTitle(filmRecord: FilmRecord) -> None:
    print(filmRecord.filmTitle)

def outputPerformerDetails(actorRecord: PerformerData) -> None:
    print(f"Performer: ${actorRecord.performerName}, Age: {actorRecord.performerAge}")

def incrementFilmCount(actorInfo: PerformerData) -> PerformerData:
    actorInfo.filmographyCount = actorInfo.filmographyCount + 1
    return actorInfo